### PR TITLE
feat: silo oracle

### PR DESCRIPTION
**Pull request by sogipec**

### CODE REVIEW

File: deploy/newOracle/newOracle.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const argv = yargs.env('').boolean('ci').parseSync();
+
+const func: DeployFunction = async ({ deployments, ethers, network }) => {
+  const { deploy } = deployments;
+  const { deployer } = await ethers.getNamedSigners();
+
+  // const treasury = (await deployments.get(`Treasury`)).address;
+  // Treasury_EUR Arbitrum
+  const treasury = '0x37963F10245e7c3a10c0E9d43a6E617B4Bc8440A';
```
Comment (GuillaumeNervoXS) [inline deploy/newOracle/newOracle.ts:14]: Just checked this address and it is notthe treasury, it is the Chainlink feed - which has a odd implementation btw

File: deploy/newOracle/newOracle.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  console.log('Now deploying the Oracle stEUR/ETH');
+  console.log(`Treasury: ${treasury}`);
+  await deploy('Oracle_STEUR_ETH', {
+    contract: `OracleSTEURETHChainlinkArbitrum`,
+    from: deployer.address,
+    args: [3600 * 36, treasury],
```
Comment (GuillaumeNervoXS) [inline deploy/newOracle/newOracle.ts:20]: The heartbeat needed for Silo?

File: contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        return _circuitChainlink;
+    }
+
+    /// @inheritdoc BaseOracleChainlinkMultiTwoFeeds
+    function _getQuoteAmount() internal view override returns (uint256) {
+        return STEUR.convertToAssets(1 ether);
+    }
+
+    /// @notice Returns the decimals of the oracle
+    function decimals() external pure returns (uint8) {
```
Comment (GuillaumeNervoXS) [inline contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol:35]: Ok this is also for Chainlink compatibility?

File: contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /// @notice Chainlink interface compatibility
+    /// @return roundID
+    /// @return aggregatorPrice
+    /// @return startedAt
+    /// @return timestamp
+    /// @return answeredInRound
+    function latestRoundData() external view returns (uint80, int256, uint256, uint256, uint80) {
+        return (0, int256(read()), 0, block.timestamp, 0);
```
Comment (GuillaumeNervoXS) [inline contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol:46]: we should just confirm with Silo if they rely on roundIds, it should be deprecated but we never know

Comment (sogipec) [reply contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol:46]: They don't, I checked their contract

File: contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol (hunk 3 above)
Comment (sogipec) [reply contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol:35]: Yes Chainlink compatibility

File: deploy/newOracle/newOracle.ts (hunk 2 above)
Comment (sogipec) [reply deploy/newOracle/newOracle.ts:20]: Yes

File: contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    /// @return roundID
+    /// @return aggregatorPrice
+    /// @return startedAt
+    /// @return timestamp
+    /// @return answeredInRound
+    /// @dev The roundId, startedAt and answeredInRound values return in this function must be disregarded
+    function latestRoundData() public view returns (uint80, int256, uint256, uint256, uint80) {
+        return (0, int256(read()), 0, block.timestamp, 0);
```
Comment (GuillaumeNervoXS) [inline contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol:57]: on the timestamp, maybe to fit their framework, we should set it to the minimum of the chainlink ones?

File: contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+
+import "../../../BaseOracleChainlinkMultiTwoFeeds.sol";
+import "../../../../interfaces/external/IERC4626.sol";
+
+/// @title OracleSTEURETHChainlinkArbitrum
+/// @author Angle Labs, Inc.
+/// @notice Gives the price of stEUR in ETH in base 18
+contract OracleSTEURETHChainlinkArbitrum is BaseOracleChainlinkMultiTwoFeeds, AggregatorV3Interface {
```
Comment (GuillaumeNervoXS) [inline contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol:13]: But we don't implement most function of AggregatorV3Interface

Comment (sogipec) [reply contracts/oracle/implementations/arbitrum/EUR/OracleSTEURETHChainlinkArbitrum.sol:13]: we do, check the interface how it's defined

**Review by GuillaumeNervoXS — APPROVED** [review]

**Pull request merged**